### PR TITLE
Fix RowValue which can reference an invalid Ok that defined outside

### PR DESCRIPTION
**Pull request by importcjj**

### CODE REVIEW

**Comment by kubo** [comment]
Thanks!

Could you change `tests/procmacro.rs` as follows?
1. Rename `fn test_procmacro_ok_shadow()` to `fn procmacro_ok_shadow()`
2. Add `#[allow(dead_code)]` to `fn procmacro_ok_shadow()` to suppress `warning: function is never used`
3. Add `#[allow(unused_imports)]` to `use AnotherResult::Ok;` to suppress `warning: unused import`
4. Run `cargo fmt`

File: tests/procmacro.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+use oracle::RowValue;
+
+#[test]
+fn procmacro_ok_shadow() {
+    #[allow(dead_code)]
+
+    enum AnotherResult<T, E> {
```
Comment (kubo) [inline tests/procmacro.rs:7]: Could you remove the empty line between `#[allow(dead_code)]` and `enum AnotherResult<T, E>`?

The attribute `#[allow(dead_code)]` qualifies `enum AnotherResult`. So the attribute should be followed by the enum without blank lines.

---
What I intended [here](https://github.com/kubo/rust-oracle/pull/62#issuecomment-1195513698) is:
```rust
#[allow(dead_code)]
fn procmacro_ok_shadow() {
    enum AnotherResult<T, E> {
```
However the following code is also acceptable.
```rust
#[test]
fn procmacro_ok_shadow() {
    #[allow(dead_code)]
    enum AnotherResult<T, E> {
```

**Comment by kubo** [comment]
@importcjj 
After you removes the empty line between `#[allow(dead_code)]` and `enum AnotherResult<T, E>`, could you merge your commits into one by `git rebase -i` (see [here](https://opensource.com/article/20/4/git-rebase-i#introducing-git-rebase--i)) and `git push -f` in order to simplify revision history?

This is optional. If you don't, I'll use "squash and merge" instead of "merge commit" (default option). (see [here](https://docs.github.com/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges))

**Comment by importcjj** [comment]
I removed the empty line， actual it doesn't take any effect.

**Comment by kubo** [comment]
Thanks for contribution.

I'll release 0.5.6 this weekend or the next weekend.

**Pull request merged**